### PR TITLE
Do not fail to start the daemon for a single backend setup failure

### PR DESCRIPTION
**Pull request by hughsie**
Type of pull request:
- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [X] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: src/fu-engine.c [hunk 1]
```diff
@@ -6385,8 +6387,13 @@ fu_engine_load (FuEngine *self, FuEngineLoadFlags flags, GError **error)
 	/* set up backends */
 	for (guint i = 0; i < self->backends->len; i++) {
 		FuBackend *backend = g_ptr_array_index (self->backends, i);
-		if (!fu_backend_setup (backend, error))
-			return FALSE;
+		g_autoptr(GError) error_backend = NULL;
```
Comment (superm1) [inline src/fu-engine.c:6390]: I think you need to special case 1 backend. If compiled with one backend it should be fatal no?

Comment (hughsie) [reply src/fu-engine.c:6390]: Would it be fair to say that we should return a failure if they _all_ fail? In the case of win32 with just gusb we'd fail as you suggest, and for Linux embedded it's unlikely we'd fail on embedded without bluez.

Comment (superm1) [reply src/fu-engine.c:6390]: Yeah I think that's reasonable.

Comment (hughsie) [reply src/fu-engine.c:6391]: Okay, something like that^^?

**Review by superm1 — APPROVED** [review]

**Pull request merged**